aspnetcore/mvc/views/tag-helpers/builtin/resources/CacheTagHelper.md

```diff
@@ -0,0 +1,286 @@
+
+Usage Example:
+
+```html
+<Cache vary-by-user="true">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+> [!NOTE]
+>  Using the attribute `vary-by-user` maintains the contents in cache through a log-in and log-out cycle.  When using `vary-by-cookie` which references the `.AspNetCore.Identity.Application` as shown above, a log-in and log-out action invalidates the cache for the same authenticated user because a new cookie value is generated. If no user is authenticated, that state is considered a valid cache state, which means that no logged-in user is one cache state, and the contents will be maintained for that condition as well.
+
+- - -
+
+### vary-by
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| String             | "@Model"  	            |
+
+
+The attribute `vary-by` allows for customization of what data gets cached. When the object referenced by the attribute's string value changes, the content of the Cache Tag Helper is updated. Often a string-concatenation of model values are assigned to this attribute.  Effectively, that means any update to any of the concatenated values invalidates the cache.
+
+The example below assumes the controller method rendering the view sums the integer value of the two route parameters, `myParam1` and `myParam2`, and returns that as the single model property. When this sum changes, the content of the Cache Tag Helper is rendered again, and the cache is reset.  
+
+Usage Example:
+
+
+*Controller*
+
+```csharp
+public IActionResult Index(string myParam1,string myParam2,string myParam3)
+{
+    int num1;
+    int num2;
+    int.TryParse(myParam1, out num1);
+    int.TryParse(myParam2, out num2);
+    return View(viewName, num1 + num2);
+}
+```
+
+*Index.cshtml*
+
+```html
+<Cache vary-by="@Model"">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### priority
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| CacheItemPriority  | "High"   	            |
+|                    | "Low" |
+|                    | "NeverRemove" |
+|                    | "Normal" |
+
+The attribute `priority` provides cache ejection guidance to the built-in cache provider. If the web server experiences memory pressure and evicts one or more of your cached entities, it will evict Low priority cache entities first.
+
+Usage Example:
+
+```html
+<Cache priority="High">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+> [!WARNING]
+>  using the `priority` attribute does not guarantee any specific level of cache retention. It is simply a suggestion to the Cache Provider. To be extra clear, setting this attribute to `NeverRemove` does not guarantee that the cache will not be evicted.  More details can be found in the additional resources listed below.
```

Comment:
* Use the contraction "It's simply ..."
* Use the contraction "doesn't" over "does not"
* I think it should be "a specific level"
* I think "Cache Provider" here would be lowercase (not a proper noun)
* Two spaces before "More"
* Stray space at the start before "using" and it should be capitalized "Using"

For this ...

> To be extra clear, setting this attribute to `NeverRemove` does not guarantee that the cache will not be evicted.

mmm ... "to be" ... they'll allow it. It makes me cringe a little bit. The "extra" is like saying "very" imo. Several style manuals call for dropping "very." I think the thinking is that it conveys little (or no) additional information.

Otherwise tho ... the sentence is a "double negative" expression. As an alternate suggestion ...

Using the `priority` attribute doesn't guarantee a specific level of cache retention. It's simply a suggestion to the cache provider. Setting this attribute to `NeverRemove` doesn't guarantee cache item persistence. For more information, see the additional resources listed below.